flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -195,4 +195,10 @@
 	 * @return Future which is completed with the {@link TransientBlobKey} of the uploaded file.
 	 */
 	CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, @RpcTimeout Time timeout);
+
+	/**
+	 * Checks whether this can be released. This cannot be released if there're unconsumed result partitions
+	 * @return whether this can be released
+	 */
+	boolean canBeReleased();
```

Comment:
I think it is better to take CompletableFuture as return type for handling timeout or failure properly.
@tillrohrmann has the authority for this.